redfish/provider/common.go

```diff
@@ -254,3 +256,28 @@ func (p powerOperator) PowerOperation(resetType string, maximumWaitTime int64, c
 	tflog.Warn(p.ctx, "The system failed to update the server's power status within the maximum wait time specified!")
 	return system.PowerState, nil
 }
+
+// checkServerStatus checks iDRAC server status after provided interval until the provided timeout time
+func checkServerStatus(ctx context.Context, endpoint string, interval int, timeout int) error {
+	var err error
+	addr, err := url.Parse(endpoint)
+	if err != nil {
+		return err
+	}
+
+	// Intial sleep until iDRAC reboot is triggered
+	time.Sleep(30 * time.Second)
+
+	for start := time.Now(); time.Since(start) < (time.Duration(timeout) * time.Second); {
+		tflog.Trace(ctx, "Checking server status...")
+		time.Sleep(time.Duration(interval) * time.Second)
+		_, err = net.Dial("tcp", addr.Hostname()+":"+addr.Scheme)
```

Comment:
addr.Scheme, in our case, would be "http" or "https". Please confirm if we really trying to create a TCP connection at "<ip_address>:https"...

Reply:
FYI, If u want ip_address:port, u can get it from addr.Host

Reply:
yes, we are making tcp connection and required format is 
Dial("tcp", "192.0.2.1:http")

addr.Host gives host i.e just ip in our case. it should be ip:port in the address when we can get the same through addr.Host. I have made use of JoinHostPort which joins host and port string.